runtime/src/iree/tooling/trace_replay.c

```diff
@@ -543,14 +547,77 @@ static void iree_trace_replay_write_element(
 }
 
 // Simple deterministic pseudorandom generator.
-// Typically in tests we want reproducible results both across runs and across
-// machines.
-static uint8_t iree_trace_replay_pseudorandom_uint8(uint32_t* state) {
-  // Same as C++'s std::minstd_rand.
+// This function is same as C++'s std::minstd_rand.
+static uint32_t iree_tree_replay_pseudorandom_uint32(uint32_t* state) {
   *state = (*state * 48271) % 2147483647;
+  return *state;
+}
+
+// Returns a random uint8_t in the range of [0, UCHAR_MAX].
+static uint8_t iree_trace_replay_pseudorandom_uint8(uint32_t* state) {
   // return the second-least-signicant out of the 4 bytes of state. it avoids
   // some mild issues with the least-significant and most-significant bytes.
-  return *state >> 8;
+  return iree_tree_replay_pseudorandom_uint32(state) >> 8;
+}
+
+// Returns a random uint32_t in the range [0, range).
+static uint32_t iree_trace_replay_pseudorandom_range(uint32_t* state,
+                                                     uint32_t range) {
+  return iree_tree_replay_pseudorandom_uint32(state) % (range);
+}
+
+// Returns a random double in the range of [0, 1.0).
+static double iree_trace_replay_pseudorandom_double(uint32_t* state) {
+  return ((double)iree_tree_replay_pseudorandom_uint32(state)) / 2147483647;
```

Comment:
if we keep this, lets introduce a named constant so this stays in sync with the PRNG implementation